src/test/java/seedu/address/model/expense/ExpenseTest.java

```diff
@@ -1,4 +1,87 @@
 package seedu.address.model.expense;
 
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_AMOUNT_BUILD_A_BEAR;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_DESCRIPTION_BUILD_A_BEAR;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_EXPENSE_CATEGORY_TRANSPORT;
+import static seedu.address.testutil.TypicalExpenses.ANNUAL_SPOTIFY;
+import static seedu.address.testutil.TypicalExpenses.BUILD_A_BEAR;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.testutil.ExpenseBuilder;
+
 public class ExpenseTest {
+
+
+    @Test
+    public void isSameExpense() {
+        // same object -> returns true
+        assertTrue(ANNUAL_SPOTIFY.isSameExpense(ANNUAL_SPOTIFY));
+
+        // null -> returns false
+        assertFalse(ANNUAL_SPOTIFY.isSameExpense(null));
+
+        // different description, all other attributes same -> returns false
+        Expense editedSpotify = new ExpenseBuilder(ANNUAL_SPOTIFY).withDescription(VALID_DESCRIPTION_BUILD_A_BEAR)
+                .build();
+        assertFalse(ANNUAL_SPOTIFY.isSameExpense(editedSpotify));
+
+        // different amount, all other attributes same -> returns false
+        editedSpotify = new ExpenseBuilder(ANNUAL_SPOTIFY).withAmount(VALID_AMOUNT_BUILD_A_BEAR).build();
+        assertFalse(ANNUAL_SPOTIFY.isSameExpense(editedSpotify));
+
+        // different expense category, all other attributes same -> returns false
+        editedSpotify = new ExpenseBuilder(ANNUAL_SPOTIFY).withExpenseCategory(VALID_EXPENSE_CATEGORY_TRANSPORT)
+                .build();
+        assertTrue(ANNUAL_SPOTIFY.isSameExpense(editedSpotify));
+
+        // name differs in case, all other attributes same -> returns false
+        Expense editedBuildABear = new ExpenseBuilder(BUILD_A_BEAR)
+                .withDescription(VALID_DESCRIPTION_BUILD_A_BEAR.toLowerCase()).build();
+        assertFalse(BUILD_A_BEAR.isSameExpense(editedBuildABear));
+
+        // TODO: date differs in case, all other attributes same -> returns false
+
+        // name has trailing spaces, all other attributes same -> returns false
+        String nameWithTrailingSpaces = VALID_DESCRIPTION_BUILD_A_BEAR + " ";
+        editedBuildABear = new ExpenseBuilder(BUILD_A_BEAR).withDescription(nameWithTrailingSpaces).build();
+        assertFalse(BUILD_A_BEAR.isSameExpense(editedBuildABear));
```

Comment:
Should this be the correct response? Parser trims the description before passing the argument to constructor right? @talkintomato

Reply:
Oh I know why already, because u're using ExpenseBuilder, but means in this case should we test this case because the parser won't in such a way.

Parser trims the input before calling for constructor of Expense class, so such a case won't happen because in the there will never be a case of a name and another which differs by a trailing space.

(Question of whether this test is essential)?

Reply:
this is to make sure the parser is doing its job which is why It is essential. same as above i think

Reply:
In this case, Expense builder don't use parser, expensebuilder is a testcase class, it is not use in the operation of the application, not sure whether essential